legal-api/src/legal_api/services/document_meta.py

```diff
@@ -0,0 +1,154 @@
+# Copyright © 2020 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This provides the service for filings documents meta data."""
+
+from enum import Enum
+
+from legal_api.models import Filing
+from legal_api.utils.legislation_datetime import LegislationDatetime
+
+
+class DocumentMetaService():
+    """Provides service for document meta data."""
+
+    class DocumentType(Enum):
+        """Define an enum of document types."""
+
+        REPORT = 'REPORT'
+
+    class ReportType(Enum):
+        """Define an enum of report types."""
+
+        CERTIFICATE = 'certificate'
+        NOTICE_OF_ARTICLES = 'noa'
```

Comment:
Please add the other 2 report types.

Also I wonder whether we should be more specific about the entity type (BC, CP, etc), for future?

Reply:
These are for capturing the required ?type query param for IA is why I added these.

@lekshmimallika-aot are there any types I missed?

Reply:
In future, when we support other incorp entity types, we may different types of certificates or notices. Can we update the code then without affecting the UI? (I don't want to add an unnecessary dependency if we can preempt it now with unique type names.)

Reply:
I think you can use the same type such as ```noa``` and ```certificate``` as it would be based on the filing itself and we can derive a Notice of Article based on the entity type when we return the output.